--- a/files/mysql.sh.orig	2014-11-17 12:58:43.000000000 +0100
+++ b/files/mysql.sh		2014-11-17 13:01:20.000000000 +0100
@@ -44,13 +44,18 @@
 PORT=$(echo $ADDR | cut -s -d ':' -f 2)
 PORT=${PORT:-"3306"}
 
+if [ -f /etc/default/glb ]; then
+    . /etc/default/glb
+fi
+WATCHDOG_CRED=${WATCHDOG_CRED:-""}
+
 QUERY="SHOW STATUS LIKE 'wsrep_local_state'; SHOW STATUS LIKE 'wsrep_incoming_addresses'"
 
 while read CMD
 do
     [ "$CMD" != "poll" ] && break;
 
-    RES=$(mysql -B --disable-column-names -h$HOST -P$PORT $* -e "$QUERY")
+    RES=$(mysql -B --disable-column-names -h$HOST -P$PORT $WATCHDOG_CRED $* -e "$QUERY")
 
     if [ $? -eq 0 ]
     then
